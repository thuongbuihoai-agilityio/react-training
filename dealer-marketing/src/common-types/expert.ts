export interface Experts {
  expertId: string;
  name: string;
  slug: string;
  info: string;
  image: {
    url: string;
    alt: string;
  };
  description: string;
}
