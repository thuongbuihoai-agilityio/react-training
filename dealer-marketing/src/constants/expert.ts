import { Expert } from "@common-types/expert";

const EXPERT_MOCKING: Expert = {
  expertId: "1",
  name: "Matt-Childers",
  slug: "matt-childers",
  info: "DealersLink",
  image: {
    url: "/images/avatar/matt-childers.png",
    alt: "This is image of Mr Matt-Childers",
  },
  description:
    "Brent Albrecht is currently the Vice President of Business Development at Friendemic. Friendemic provides social media and online reputation services for clients and agencies across the globe. Specialties: Marketing Strategy, Product development and launch, Social Media Marketing, Business Development, Lead Generation, Sales Promotion, Digital Marketing, Database Marketing.",
};

export { EXPERT_MOCKING };
