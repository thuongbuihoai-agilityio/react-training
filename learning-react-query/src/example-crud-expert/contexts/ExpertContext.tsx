import { createContext, useState } from "react";
import { Expert, ExpertContextProps } from "../interfaces/expert";
import { useExpertStore } from "../stores/expert";

const ExpertContext = createContext<ExpertContextProps>(
  {} as ExpertContextProps,
);
const ExpertProvider: React.FC<{ children: JSX.Element[] | JSX.Element }> = ({
  children,
}) => {
  const { experts, setExperts } = useExpertStore();
  const [errorCode, setErrorCode] = useState<number>(0);

  const handleUpdateExperts = (errorCode: number, experts: Expert[]) => {
    if (errorCode) {
      setErrorCode(errorCode);
    } else {
      setExperts(experts);
    }
  };

  const value = {
    experts,
    errorCode,
    setExperts,
    setErrorCode,
    handleUpdateExperts,
  };

  return (
    <ExpertContext.Provider value={value}>{children}</ExpertContext.Provider>
  );
};

export { ExpertProvider, ExpertContext };
