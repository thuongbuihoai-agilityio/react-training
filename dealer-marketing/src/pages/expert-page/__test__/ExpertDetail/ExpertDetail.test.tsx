import { EXPERT_MOCKING, EXPERT_MOCKING_LIST } from "@constants/expert";
import { ExpertContextProps } from "@common-types/expert";
import { ExpertContext } from "@context/ExpertContext";
import { render } from "@testing-library/react";
import OurExpert from "@pages/expert-page/[slug]";

const contextExpertMocking: ExpertContextProps = {
  experts: EXPERT_MOCKING_LIST,
  errorCode: 0,
  setExperts: jest.fn(),
  setErrorCode: jest.fn(),
  handleUpdateExperts: jest.fn(),
};

describe("ExpertDetail page", () => {
  test("Should render ExpertDetail page", () => {
    const { getByTestId } = render(
      <ExpertContext.Provider value={contextExpertMocking}>
        <OurExpert expert={EXPERT_MOCKING} />
      </ExpertContext.Provider>,
    );
    const expertDetail = getByTestId("layout");
    expect(expertDetail).toBeInTheDocument();
  });

  test("Matches snapshot", () => {
    const { container } = render(
      <ExpertContext.Provider value={contextExpertMocking}>
        <OurExpert expert={EXPERT_MOCKING} />
      </ExpertContext.Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
