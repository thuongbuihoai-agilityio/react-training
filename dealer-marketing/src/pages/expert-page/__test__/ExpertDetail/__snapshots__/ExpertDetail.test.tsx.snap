// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpertDetail page Matches snapshot 1`] = `
<div>
  <main
    data-testid="layout"
  >
    <header
      data-testid="header"
    >
      <div>
        <ul
          data-testid="menu"
        >
          <li>
            <a
              href="/expert-panel"
            >
              Our Experts
            </a>
          </li>
          <li>
            <a
              href="/about-us"
            >
              About
            </a>
          </li>
        </ul>
      </div>
    </header>
    <nav
      data-testid="navigation"
    >
      <div
        data-testid="logo"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27256%27%20height=%2743%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="This is logo"
            data-nimg="intrinsic"
            decoding="async"
            href="/"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(/images/backgrounds/blur.jpg);"
          />
          <noscript />
        </span>
      </div>
      <div>
        <ul
          data-testid="menu"
        >
          <li>
            <a
              href="/"
            >
              Research
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              Best practice
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              Commentary
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              Reviews
            </a>
          </li>
        </ul>
        <i
          data-testid="icon"
        />
      </div>
    </nav>
    <div
      data-testid="banner"
    >
      <div>
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="This is image of banner"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(/images/backgrounds/blur.jpg);"
          />
          <noscript />
        </span>
      </div>
      <h1 />
    </div>
    <div>
      <figure>
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27130%27%20height=%27130%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="This is image of Mr Matt-Childers"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(/images/backgrounds/blur.jpg);"
          />
          <noscript />
        </span>
      </figure>
      <div>
        <h2>
          Matt-Childers
        </h2>
        <div>
          <p
            data-testid="text"
          >
            DealersLink
          </p>
        </div>
        <p
          data-testid="text"
        >
          Brent Albrecht is currently the Vice President of Business Development at Friendemic. Friendemic provides social media and online reputation services for clients and agencies across the globe. Specialties: Marketing Strategy, Product development and launch, Social Media Marketing, Business Development, Lead Generation, Sales Promotion, Digital Marketing, Database Marketing.
        </p>
      </div>
    </div>
    <footer
      data-testid="footer"
    >
      <ul
        data-testid="menu"
      >
        <li>
          <a
            href="/expert-panel"
          >
            Our Experts
          </a>
        </li>
        <li>
          <a
            href="/about-us"
          >
            About
          </a>
        </li>
      </ul>
      <div>
        <ul
          data-testid="menu"
        >
          <li>
            <a
              href="/"
            >
              Privacy policy
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              Terms of service
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              Advertise
            </a>
          </li>
        </ul>
        <p>
          Copyright Â© 2022 Dealer Marketing Magazine
        </p>
      </div>
    </footer>
  </main>
</div>
`;
